cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/ApiConnectionInfo.java

```diff
@@ -25,27 +25,48 @@
 package mil.army.usace.hec.cwms.http.client;
 
 import java.util.Optional;
-import mil.army.usace.hec.cwms.http.client.auth.OAuth2Token;
+import mil.army.usace.hec.cwms.http.client.auth.OAuth2TokenProvider;
 
 public final class ApiConnectionInfo {
 
     private final String apiRoot;
-    private final OAuth2Token oauth2Token;
+    private final OAuth2TokenProvider tokenProvider;
+    private final SslSocketData sslSocketData;
 
     public ApiConnectionInfo(String apiRoot) {
-        this(apiRoot, null);
+        this(apiRoot, null, null);
     }
 
-    public ApiConnectionInfo(String apiRoot, OAuth2Token oauth2Token) {
+    public ApiConnectionInfo(String apiRoot, SslSocketData sslSocketData) {
+        this(apiRoot, sslSocketData, null);
+    }
+
+    public ApiConnectionInfo(String apiRoot, OAuth2TokenProvider tokenProvider) {
```

Comment:
More than 2 constructors triggers my "this should have a build instead" instinct.

Reply:
Updating to use a builder.